From: unknown author <unknown.author@example.com>
Date: Wed, 28 Jan 2015 19:02:25 +0100
Subject: [PATCH] Handle the visible size of a display in the kernel

Instead of guessing the visible size of the graphic area at
run-time, set it up in the kernel with the data from its datasheet

Signed-off-by: Juergen Beisert <jbe@pengutronix.de>
---
 arch/arm/plat-samsung/include/plat/fb-s3c2410.h | 2 ++
 drivers/video/fbdev/s3c2410fb.c                 | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-samsung/include/plat/fb-s3c2410.h b/arch/arm/plat-samsung/include/plat/fb-s3c2410.h
index 4e5d9588b5ba..75e85a2acce4 100644
--- a/arch/arm/plat-samsung/include/plat/fb-s3c2410.h
+++ b/arch/arm/plat-samsung/include/plat/fb-s3c2410.h
@@ -33,6 +33,8 @@ struct s3c2410fb_display {
 	unsigned short xres;
 	unsigned short yres;
 	unsigned short bpp;
+	unsigned int phys_width; /* visible width in [mm] */
+	unsigned int phys_height; /* visible height in [mm] */
 
 	unsigned pixclock;		/* pixclock in picoseconds */
 	unsigned short left_margin;  /* value in pixels (TFT) or HCLKs (STN) */
diff --git a/drivers/video/fbdev/s3c2410fb.c b/drivers/video/fbdev/s3c2410fb.c
index 2a148b1dd8aa..27608119fe25 100644
--- a/drivers/video/fbdev/s3c2410fb.c
+++ b/drivers/video/fbdev/s3c2410fb.c
@@ -157,8 +157,8 @@ static int s3c2410fb_check_var(struct fb_var_screeninfo *var,
 	/* it is always the size as the display */
 	var->xres_virtual = display->xres;
 	var->yres_virtual = display->yres;
-	var->height = display->height;
-	var->width = display->width;
+	var->height = display->phys_height;
+	var->width = display->phys_width;
 
 	/* copy lcd settings */
 	var->pixclock = display->pixclock;
